code/game/machinery/computer/apc_control.dm

```diff
@@ -0,0 +1,164 @@
+/obj/machinery/computer/apc_control
+	name = "power flow control console"
+	desc = "Used to remotely control the flow of power to different parts of the station."
+	icon_screen = "solar"
+	icon_keyboard = "power_key"
+	circuit = /obj/item/weapon/circuitboard/computer/apc_control
+	light_color = LIGHT_COLOR_YELLOW
+	var/list/apcs //APCs the computer has access to
+	var/mob/living/operator //Who's operating the computer right now
+	var/obj/machinery/power/apc/active_apc //The APC we're using right now
+	var/list/filters //For sorting the results
+	var/checking_logs = 0
+	var/list/logs
+
+/obj/machinery/computer/apc_control/Initialize()
+	apcs = list() //To avoid BYOND making the list run through a ton of procs
+	filters = list("Name" = null, "Charge Above" = null, "Charge Below" = null, "Responsive" = null)
+	..()
+
+/obj/machinery/computer/apc_control/process()
+	apcs = list() //Clear the list every tick
+	for(var/V in apcs_list)
+		var/obj/machinery/power/apc/APC = V
+		if(check_apc(APC))
+			apcs[APC.name] = APC
+	if(operator)
+		if(!operator.Adjacent(src))
```

Comment:
If you do decide to allow silicon control, please remember that you need to check for 'operator.has_unlimited_silicon_privilege' or 'issilicon(operator)` (Only one, not both) such that they are not locked out by the Adjacent check.

If you do NOT want silicons to control the console, then override `attack_ai()` with a simple `return` such that you do not have oddities like putting an AI directly next to the console, or a borg simply rolling up to it.